### ISSUE TITLE
Symmetric sparse matrices

### ISSUE THREAD

**Issue by dswatson** [yixuan/RSpectra#24]
Thanks for this great package! Was wondering if you could help with a issue.

The documentation suggests that `eigs_sym()` should be able to handle sparse matrix input, but there appears to be some subtleties to handling sparsity in symmetric matrices. Specifically, I get an error when I run this code:

```r
# Create sparse, symmetric matrix
library(Matrix)
set.seed(1)
n <- 4
M <- matrix(0, nrow = n, ncol = n)
M[sample(n, 2), sample(n, 2)] <- rnorm(n)
A <- crossprod(M)
A <- as(A, 'sparseMatrix') # Note the class of A is dsCMatrix

# This works
e <- eigs(A, k = 2)

# This does not
e <- eigs_sym(A, k = 2)
```

The last line generates the following error: 

```
Error in UseMethod("eigs_sym") : 
  no applicable method for 'eigs_sym' applied to an object of class "c('dsCMatrix', 'CsparseMatrix', 'dsparseMatrix', 'symmetricMatrix', 'dMatrix', 'sparseMatrix', 'Matrix')"
```

Interestingly, I do not get the same error when passing a `dgeMatrix` to `eigs_sym`. Hopefully this is an easy fix?

**Comment by yixuan** [comment]
Yeah, this is actually a design choice. `eigs()` tries to automatically detect the type of the matrix, and if it is `dsCMatrix`/`dsRMatrix`/`dsyMatrix`, it will directly call the symmetric solver. On the other hand, `eigs_sym()` in some sense "forces" the symmetry of the matrix, so I did not let it work on matrices that are already symmetric.

**Comment by dswatson** [comment]
That makes sense, thanks! Resolves my issue. Perhaps this could be a bit clearer in the documenation? Or maybe instead of an error, `eigs_sym()` could just run `eigs()` internally when the input is of class `dsCMatrix`?

**Comment by yixuan** [comment]
Yeah, good suggestion. This is actually documented in the help, but I agree that it can be made clearer.

![image](https://github.com/user-attachments/assets/4956f7fb-0643-48ba-b39e-fc43704a3494)

I may change it in the next release.

**Comment by yixuan** [comment]
The support for symmetric matrix types in `eigs_sym()` has been added in https://github.com/yixuan/RSpectra/commit/3b7d3ce0e6cd1aa3079ade2d237ada81290dd471 and https://github.com/yixuan/RSpectra/commit/ef6b3548c792487ab446340b5b8f33cd798b31c3.
